invisible_cities/reco/deconv_functions.py

```diff
@@ -305,11 +373,33 @@ def richardson_lucy(image, psf, iterations=50, iter_thr=0.):
         relative_blur = image / x
         im_deconv *= convolve_method(relative_blur, psf_mirror, 'same')
 
+        # after every iteration, kill satellites
+        if (i > satellite_iter):
+            # apply mask to a copy
+            im_vis = im_deconv.copy()
+            # set based on relative or absolute cut
+            if cut_type == CutType.abs:
+                vis_mask = im_vis
+            elif cut_type == CutType.rel:
+                vis_mask = im_vis / im_vis.max()
+            else:
+                raise ValueError(f'cut_type {cut_type} is not a valid cut type.')
+
+            # apply the mask cut
+            im_vis[vis_mask < e_cut] = 0
+            im_vis[vis_mask >= e_cut] = 1
+            # create mask that removes satellites
+            satellite_mask = isolate_satellites(im_vis, satellite_dist, satellite_size)
+            # remove only satellite regions!
+            deconv_mask = (im_vis + satellite_mask) % 2 == 0
```

Comment:
Explain a bit more

Reply:
This line is gone as the logic behind this part of the code was messy.